VAULT-36949: Automated snapshot cloud permissions (#1145)
hashicorp/web-unified-docs: content/vault/v1.16.x/content/api-docs/system/storage/raftautosnapshots.mdx

- `aws_s3_force_path_style` `(boolean)` - Use the endpoint/bucket URL style
  instead of bucket.endpoint. May be needed when setting `aws_s3_endpoint`.

> - `aws_s3_enable_kms` `(boolean)` - Use KMS to encrypt bucket contents. If this is true, make sure that your AWS account also has the permissions:

Reviewer: ```suggestion
- `aws_s3_enable_kms` `(boolean)` - Use KMS to encrypt bucket contents. To use
   KMS encryption, the associated AWS account must have the following
   permissions:
```
Style correction: avoid "this" as a pronoun

- `google_service_account_key` `(string)` - Google service account key in JSON format. Depending
  on how the API is invoked, this may be need to be JSON-escaped, e.g. for newlines and double quotes.
  The raw value looks like this:
  ```json
  {
    "type": "service_account",
    "project_id": "project-id",
    "private_key_id": "key-id",
    "private_key": "-----BEGIN RSA PRIVATE KEY-----\nMIIEpQ ... /WZs=\n-----END RSA PRIVATE KEY-----\n",
    "client_email": "service-account-email",
    "client_id": "client-id",
    "auth_uri": "https://accounts.google.com/o/oauth2/auth",
    "token_uri": "https://accounts.google.com/o/oauth2/token",
    "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
    "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/service-account-email"
  }
    ```
  

>   Make sure that the Google service account used has the following permissions in Google Cloud Storage, so that Vault can store and manage the snapshots:

Reviewer: ```suggestion
  Your Google service account must have the following permissions in Google Cloud Storage so that Vault can store and manage the snapshots:
```
Style correction: state requirements plainly

- `azure_container_name` `(string: <required>)` - Azure container name to write
  snapshots to.

- `azure_account_name` `(string)` - Azure account name. Make sure that this account has

>   the following permissions in Azure Blob Storage, so that Vault can store and manage the snapshots:

Reviewer: ```suggestion
- `azure_account_name` `(string)` - Azure account name. The associated Azure
  account must have the following permissions in Azure Blob Storage so that
  Vault can store and manage the snapshots:
```
